src/dmd/dtemplate.d

```diff
@@ -38,43 +38,54 @@
 
 module dmd.dtemplate;
 
-import core.stdc.stdio;
-import core.stdc.string;
-import dmd.aggregate;
-import dmd.arraytypes;
-import dmd.ast_node;
-import dmd.dcast;
-import dmd.dclass;
-import dmd.declaration;
-import dmd.dmangle;
-import dmd.dmodule;
-import dmd.dscope;
-import dmd.dsymbol;
-import dmd.dsymbolsem;
-import dmd.errors;
-import dmd.expression;
-import dmd.expressionsem;
-import dmd.func;
-import dmd.globals;
-import dmd.hdrgen;
-import dmd.id;
-import dmd.identifier;
-import dmd.impcnvtab;
-import dmd.init;
-import dmd.initsem;
-import dmd.mtype;
-import dmd.opover;
-import dmd.root.array;
-import dmd.root.outbuffer;
-import dmd.root.rmem;
-import dmd.root.rootobject;
-import dmd.semantic2;
-import dmd.semantic3;
-import dmd.tokens;
-import dmd.typesem;
-import dmd.visitor;
-
-import dmd.templateparamsem;
+import core.stdc.stdio : printf;
+import core.stdc.string : memcpy;
+import dmd.aggregate : AggregateDeclaration;
+import dmd.arraytypes : Objects, TemplateParameters, Modules, Expressions, Dsymbols, Types,
+    TemplateInstances, Parameters;
+import dmd.ast_node : ASTNode;
+import dmd.dcast : toStaticArrayType;
+import dmd.dclass : BaseClass, ClassDeclaration;
+import dmd.declaration : MatchAccumulator, Declaration, VarDeclaration, STC, AliasDeclaration,
+    OverDeclaration, TupleDeclaration;
+import dmd.dmangle : mangleToBuffer;
+import dmd.dmodule : Module;
+import dmd.dscope : Scope, SCOPE;
+import dmd.dsymbol : Dsymbol, ScopeDsymbol, WithScopeSymbol, Prot, DsymbolTable, PASS, Ungag,
+    OverloadSet, foreachDsymbol;
+import dmd.dsymbolsem : aliasSemantic, dsymbolSemantic, templateInstanceSemantic;
+import dmd.errors : error, errorSupplemental, fatal, message, tip;
+import dmd.expression : Expression, VarExp, IntervalExp, RealExp, ComplexExp, IdentityExp, NullExp,
```

Comment:
Maybe these import should not be selective given that the import iterates most of the symbols defined in `expression.d`.

Reply:
I had that thought too.  I'm just trying to feel out what is preferred, useful, etc.